### DISCUSSION TITLE
When we add button component to the editor , it changes to text component after a reload . Please help

### DISCUSSION THREAD

**Discussion by Sudhin35** [GrapesJS/grapesjs#3889] [Q&A]
Hi All ,

I have created a custome button component in the grapes js , when dragged and dropped , it is successfully created in the editor but after the reload the button component is changing to the text component . Please help 

I have attached the code and screenshots 

```
const newBlocksEl = blockManager.render( [
        {
           id: 'themed_button1',
          label: '<button type="button" class="btn btn-light">Button1</button>',
          type: "button",
          category: 'Themed Button',
          content: '<button type="button" class="btn [btn-light">Button1</button>',
          activate: true,
        },
```

![button block](https://user-images.githubusercontent.com/50030377/138689633-601ac178-74de-4436-a81c-e48b4fa088eb.png)

**Comment by sandenson** [comment]
You should add a Block to the BlockManager like this:

```js
blockManager.add('h1-block', {
  label: 'Heading',
  content: '<h1>Put your title here</h1>',
  category: 'Basic',
  attributes: {
    title: 'Insert h1 block'
  }
});
```

Here's the [API reference](https://grapesjs.com/docs/api/block_manager.html) for the BlockManager.

**Comment by artf** [comment]
You created the Block, that doesn't have any component context for the editor in that way so it falls back to the best default option (Text in that case). Try to read [Components](https://grapesjs.com/docs/modules/Components.html) first, then [Blocks](https://grapesjs.com/docs/modules/Blocks.html).

Probably our falt is not to have all HTML elements as components (in your case the `<button>`), which is something I'd like to fix soon by creating a new dedicated plugin.
